### PR TITLE
feat: add combinators

### PR DESCRIPTION
**Pull request by Laura-Berkers**
## Description

This PR adds static methods which are wrappers around the constructors of goal structures and tactics, so the user does not have to write `new` every time.

The user should now be able to create a snippet looking something like this:
```cs
GoalStructure gs = FirstOf(Primitive(...));
Tactic t = FirstOf(AnyOf(...), AnyOf(...));
```

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch

### CODE REVIEW

**Review by joachimdekker — DISMISSED** [review]

**Review by joachimdekker — DISMISSED** [review]
Please listen to [this masterpiece](https://suno.com/song/6518084a-2ee6-4c92-8454-7dac15614afb)

**Review by joachimdekker — DISMISSED** [review]

File: Aplib.Core/Combinators.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+using Aplib.Core.Intent.Tactics;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Convenience class containing static methods for creating goal structures and tactics.
+    /// </summary>
+    public static class Combinators
+    {
+        #region GoalSructure combinators
```
Comment (JensSteenmetz) [inline Aplib.Core/Combinators.cs]: image

```suggestion
        #region GoalStructure combinators
```

**Review by JensSteenmetz — DISMISSED** [review]
My reaction to the tests:
image

**Review by JensSteenmetz — APPROVED** [review]
LGTM

**Review by joachimdekker — APPROVED** [review]

**Pull request merged**